converters/google/resources/custom_org_policy.go

```diff
@@ -0,0 +1,184 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+)
+
+func resourceConverterCustomOrgPolicy() ResourceConverter {
+	return ResourceConverter{
+		Convert:           GetCustomOrgPolicyCaiObject,
+		MergeCreateUpdate: MergeCustomOrgPolicy,
+	}
+}
+
+func GetCustomOrgPolicyCaiObject(d TerraformResourceData, config *Config) ([]Asset, error) {
+
+	assetNamePattern, assetType, err := getAssetNameAndTypeFromParent(d.Get("parent").(string))
+	if err != nil {
+		return []Asset{}, err
+	}
+
+	name, err := assetName(d, config, assetNamePattern)
+	if err != nil {
+		return []Asset{}, err
+	}
+
+	if obj, err := GetCustomOrgPolicyApiObject(d, config); err == nil {
+		return []Asset{{
+			Name:            name,
+			Type:            assetType,
+			CustomOrgPolicy: []*CustomOrgPolicy{&obj},
+		}}, nil
+	} else {
+		return []Asset{}, err
+	}
+
+}
+
+func GetCustomOrgPolicyApiObject(d TerraformResourceData, config *Config) (CustomOrgPolicy, error) {
+	spec, err := expandSpecCustomOrgPolicy(d.Get("spec").([]interface{}))
+	if err != nil {
+		return CustomOrgPolicy{}, err
+	}
+
+	return CustomOrgPolicy{
+		Name: d.Get("name").(string),
+		Spec: spec,
+	}, nil
+}
+
+func MergeCustomOrgPolicy(existing, incoming Asset) Asset {
+	existing.Resource = incoming.Resource
```

Comment:
a merge function like this is used for Project, as a workaround for the fact that there are sometimes duplicates of data for a particular project that we want to just ignore - but I wouldn't expect us to need it for any other resources. The other place we use merge functions is for IAM, which is because in Terraform, you can specify partial IAM policies (using member or binding resources) which have to be merged with the current policy from the API to find out what the actual end state will be.

I would recommend not having a merge function like this - unless you're sure that this will be necessary.

Reply:
Hmm, looking at this some more, it seems like a resource can have multiple CustomOrgPolicies attached. In that case - maybe the merge function should be appending the `incoming.CustomOrgPolicy` list to the `existing.CustomOrgPolicy` list? It could be good to have a test for this case as well.